kornia/geometry/camera/stereo.py

```diff
@@ -0,0 +1,287 @@
+import torch
+
+
+class StereoException(Exception):
+    def __init__(self, msg, *args, **kwargs):
+        """
+        Custom exception for the :module:`~kornia.geometry.camera.stereo` module. Adds a general helper module
+        redirecting the user to the proper documentation site.
+        Args:
+            msg: Custom message to add to the general message.
+            *args:
+            **kwargs:
+        """
+        doc_help = (
+            "\n Please check documents here: "
+            "https://kornia.readthedocs.io/en/latest/geometry.camera.stereo.html"
+            "for further information and examples."
+        )
+        final_msg = msg + doc_help
+        super().__init__(final_msg, *args, **kwargs)  # type: ignore
+
+
+class StereoCamera:
+    def __init__(self, rectified_left_camera: torch.Tensor, rectified_right_camera: torch.Tensor):
```

Comment:
let's target directly to instantiate from `PinholeCamera` unless there another use case I miss.

Reply:
I like this idea, but it is still unclear how to initialize the extrinsic of the `PinholeCamera` class for left and right respectively.

The most straight forward is keeping the intrinsic of each camera the same and then add the horizontal stereo offset to the extrensic camera of the right camera matrix. Pseudo code below:

```
left_int = [
[fx, 0, cx, 0],
[0, fy, cy, 0],
[0, 0, 1, 0],
[0, 0, 0, 1]]

right_int = 
[fx, 0, cx, 0],
[0, fy, cy, 0],
[0, 0, 1, 0],
[0, 0, 0, 1]]

left_ext = torch.eye(4)

right_ext = [[1, 0, 0, Tx * fx]
[0, 1, 0, 0]
[0, 0, 0, 0]
[0, 0, 0, 0]]
``` 

I think this is a bit more complicated and also breaks a little with ROS/OpenCV convention, but we could go this way.

What do you think?

Reply:
@edgarriba any thoughts on this idea?